src/memex/views.py

```diff
@@ -248,12 +248,16 @@ def read_jsonld(context, request):
 
 
 @api_config(route_name='api.annotation',
-            request_method='PUT',
+            request_method=('PATCH', 'PUT'),
             permission='update',
             link_name='annotation.update',
             description='Update an annotation')
 def update(context, request):
-    """Update the specified annotation with data from the PUT payload."""
+    """Update the specified annotation with data from the PATCH payload."""
+    if request.method == 'PUT' and hasattr(request, 'stats'):
+        stats = getattr(request, 'stats')
```

Comment:
Out of interest, what’s the reasoning behind using `getattr(request, 'stats')` here instead of `request.stats`? Is it for symmetry with the `hasattr` on the line above, or is there a difference between the two that I’ve missed?